File: server/pkg/entities/task.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Task struct {
	ID          uuid.UUID      `gorm:"type:uuid;default:uuid_generate_v4()" json:"id"`
	ColumnID    uuid.UUID      `gorm:"type:uuid;not null;"                  json:"columnId"`
	AssigneeID  *uuid.UUID     `gorm:"type:uuid"                            json:"assigneeId"`
	Title       string         `gorm:"not null;size:50;"                    json:"title"       validate:"required,min=3,max=50"`
	Description string         `gorm:"type:text;"                           json:"description" validate:"max=2000"`
	Priority    int            `gorm:"not null;"                            json:"priority"`
	DueDate     time.Time      `gorm:"type:timestamp;"                      json:"dueDate"`
	Position    int            `gorm:"not null;"                            json:"position"`
	CreatedAt   time.Time      `                                            json:"createdAt"`
	UpdatedAt   time.Time      `                                            json:"updatedAt"`
	DeletedAt   gorm.DeletedAt `gorm:"index"                                json:"deletedAt"`
}

File: client/src/schemas/task.ts
import { z } from "zod";

export const CreateTaskSchema = z.object({
  columnId: z.string().min(1),
  assigneeId: z.string().nullable(),
  position: z.number().int().min(0),
  title: z.string().min(1),
  description: z.string().nullable(),
  dueDate: z.number().nullable(),
  priority: z.number().int().min(0),
});

export const DeleteTaskSchema = z.object({
  columnId: z.string().min(1),
});

export const UpdateTaskSchema = z.object({
  columnId: z.string().min(1),
  assigneeId: z.string().nullable(),
  title: z.string().min(1),
  description: z.string().nullable(),
  dueDate: z.number().nullable(),
  priority: z.number().int().min(0).nullable(),
});

export const ChangeTaskStatusSchema = z.object({
  columnId: z.string().min(1),
  position: z.number().int().min(0),
});
